File: java/src/test/java/com/kevinthorley/csi280/project01/ClassRepoTest.java
package com.kevinthorley.csi280.project01;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

import org.junit.Test;

public class ClassRepoTest {

  @Test
  public void testFindById() 
  {
    ClassRepo classRepo = new ClassRepo();

    Section mySection = classRepo.findById("csi-280-51");

    assertNotNull(mySection);
    assertEquals("Open Source Software Development", mySection.getName());
  }

  @Test
  public void testAdd() {
    ClassRepo classRepo = new ClassRepo();

    Section newSection = new Section("csi-180", "Emerging Technologies", "2016SP");
    classRepo.addSection(newSection);

    Section mySection = classRepo.findById("csi-180");

    assertNotNull(mySection);
    assertEquals("Emerging Technologies", mySection.getName());
  }
}

File: javascript/test/classrepotest.js
var should = require('should');
var classrepo = require('../lib/classrepo');

describe('ClassRepo', function() {

  var classRepo;

  beforeEach(function() {
    classRepo = new classrepo.ClassRepo();
  });

  describe('findById', function() {
    it('should retrieve a class with the given id', function() {
      var myClass = classRepo.findById('csi-280-51');
      

      should.exist(myClass);
      myClass.name.should.equal('Open Source Software Development');
    });
  });

  describe('add', function() {
    it('should add a new class', function() {
      var newClass = {
        id: 'csi-180',
        name: 'Emerging Technologies',
        semester: '2016SP'
      };

      classRepo.add(newClass);

      var myClass = classRepo.findById('csi-180');

      should.exist(myClass);
      myClass.name.should.equal('Emerging Technologies');
    });
  });
});
